widgets/cpu_openbsd.lua

```diff
@@ -0,0 +1,60 @@
+-- cpu_openbsd.lua - provide CPU usage on OpenBSD
+-- Copyright (C) 2019  Enric Morales
+--
+-- This file is part of Vicious.
+--
+-- Vicious is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU General Public License as
+-- published by the Free Software Foundation, either version 2 of the
+-- License, or (at your option) any later version.
+--
+-- Vicious is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with Vicious.  If not, see <https://www.gnu.org/licenses/>.
+
+local math = { ceil = math.ceil }
+local string = { gmatch = string.gmatch }
+local table = { insert = table.insert }
+local tonumber = tonumber
+
+local helpers = require("vicious.helpers")
+
+
+-- cpu_openbsd: provides both a helper function that allows reading
+-- the CPU usage on OpenBSD systems.
+local cpu_openbsd = {}
+
+-- Initialize the table that will contain the ticks spent in each subsystem
+-- values: user, nice, system, spin, interrupts, idle
+cpu_openbsd.ticks = { 0, 0, 0, 0, 0, 0 }
```

Comment:
I believe using local `ticks` here would already be enough.

Reply:
I tried using `local ticks` within the `cpu_openbsd.async` function and the state wouldn't be stored 😭 for reuse when running the function again.

Reply:
I meant local ticks at the module level (closure).  While it might be obvious to you when you write it, future ready might confused on why `ticks` has to be part of the returning table.  TBH it took me a while.